system-contracts/contracts/EvmInterpreterFunctions.template.yul

```diff
@@ -776,11 +774,13 @@ function getEVMGas() -> evmGas {
     evmGas := div(sub(_gas, requiredGas), GAS_DIVISOR())
 }
 
-function _getZkEVMGas(_evmGas) -> zkevmGas {
-    /*
-        TODO: refine the formula, especially with regard to decommitment costs
-    */
+function _getZkEVMGas(_evmGas, addr) -> zkevmGas {
     zkevmGas := mul(_evmGas, GAS_DIVISOR())
+    let byteSize := extcodesize(addr)
+    zkevmGas := add(zkevmGas, mul(byteSize, DECOMMIT_COST_PER_WORD()))
```

Comment:
1. the cost is _per word_ (as the name implies). I.e. it is not `byteSize * DECOMMIT_COST_PER_WORD`, it is `ceil(byteSize / 32) * DECOMMIT_COST_PER_WORD`.
2. the cost should be charged and not added to the user.

Reply:
Fixed here [1a88e64](https://github.com/matter-labs/era-contracts/pull/481/commits/1a88e645c6c8709c4dfdb1489da0ea6026ec1364)